mecanum_drive_controller/test/test_mecanum_drive_controller.cpp

```diff
@@ -0,0 +1,592 @@
+  executor.add_node(controller_->get_node()->get_node_base_interface());
+
+  ASSERT_EQ(controller_->on_configure(rclcpp_lifecycle::State()), NODE_SUCCESS);
+  ASSERT_EQ(controller_->on_activate(rclcpp_lifecycle::State()), NODE_SUCCESS);
+
+  auto reference = controller_->input_ref_.readFromNonRT();
+  auto old_timestamp = (*reference)->header.stamp;
+  EXPECT_TRUE(std::isnan((*reference)->twist.linear.x));
+  EXPECT_TRUE(std::isnan((*reference)->twist.linear.y));
+  EXPECT_TRUE(std::isnan((*reference)->twist.angular.z));
+
+  // reference_callback() is implicitly called when publish_commands() is called
+  publish_commands(rclcpp::Time(0));
+
+  ASSERT_TRUE(controller_->wait_for_commands(executor));
+  ASSERT_EQ(old_timestamp.sec, (*(controller_->input_ref_.readFromNonRT()))->header.stamp.sec);
+  EXPECT_FALSE(std::isnan((*(controller_->input_ref_.readFromNonRT()))->twist.linear.x));
+  EXPECT_FALSE(std::isnan((*(controller_->input_ref_.readFromNonRT()))->twist.angular.z));
+  EXPECT_EQ((*(controller_->input_ref_.readFromNonRT()))->twist.linear.x, 1.5);
+  EXPECT_EQ((*(controller_->input_ref_.readFromNonRT()))->twist.linear.y, 0.0);
+  EXPECT_EQ((*(controller_->input_ref_.readFromNonRT()))->twist.angular.z, 0.0);
+  EXPECT_NE((*(controller_->input_ref_.readFromNonRT()))->header.stamp.sec, 0.0);
+}
+
+// when age_of_last_command < ref_timeout expect reference msg is accepted and is in rt buffer
+TEST_F(MecanumDriveControllerTest, when_message_accepted_expect_reference_msg_in_rt_buffer)
+{
+  SetUpController();
+
+  rclcpp::executors::MultiThreadedExecutor executor;
+  executor.add_node(controller_->get_node()->get_node_base_interface());
+
+  ASSERT_EQ(controller_->on_configure(rclcpp_lifecycle::State()), NODE_SUCCESS);
+  ASSERT_EQ(controller_->on_activate(rclcpp_lifecycle::State()), NODE_SUCCESS);
+
+  auto reference = controller_->input_ref_.readFromNonRT();
+  EXPECT_TRUE(std::isnan((*reference)->twist.linear.x));
+  EXPECT_TRUE(std::isnan((*reference)->twist.angular.z));
+
+  // reference_callback() is implicitly called when publish_commands() is called
+  publish_commands(controller_->get_node()->now());
+
+  ASSERT_TRUE(controller_->wait_for_commands(executor));
+  EXPECT_FALSE(std::isnan((*(controller_->input_ref_.readFromNonRT()))->twist.linear.x));
+  EXPECT_FALSE(std::isnan((*(controller_->input_ref_.readFromNonRT()))->twist.angular.z));
+  EXPECT_EQ((*(controller_->input_ref_.readFromNonRT()))->twist.linear.x, 1.5);
+  EXPECT_EQ((*(controller_->input_ref_.readFromNonRT()))->twist.linear.y, 0.0);
+  EXPECT_EQ((*(controller_->input_ref_.readFromNonRT()))->twist.angular.z, 0.0);
+}
+
+// when not in chainable mode and age_of_last_command > reference_timeout expect
+// command_interfaces are set to 0.0 and reference_interfaces set to nan
+// followed by
+// when not in chainable mode and age_of_last_command < reference_timeout expect
+// command_interfaces are calculated to non-nan and reference_interfaces set to nan
```

Comment:
It is not clear what we are testing here. do you want to test message timeout? Then write it as it is. This write-up nobody can understand what is actually test about.

Please update this and the whole test as well. As it is too complex, and many checks are just not needed…

Reply:
@destogl  this test title doesnt fit to the content, 
i am not even sure if i wrote this title. message timeout test is already written before with title, when_reference_msg_is_too_old_expect_unset_reference


if you remember you asked me to write a test that simulates a flow of messages, where first comes a message that has timeout and then next comes a message that is not timed out.

Reply:
ok i will change he title to what i proposed then.
i agree with the comments will change it accordingly.